structure_tests/structure_test_v1.go

```diff
@@ -121,3 +98,89 @@ func (st StructureTestv1) RunFileContentTests(t *testing.T) {
 		}
 	}
 }
+
+// currently a hack to replace PWD in output/error until we figure
+// out a better way to handle this.
+func SubstituteEnvVars(t *testing.T, lists ...*[]string) {
+	env_var := "PWD"
+	value := os.Getenv(env_var)
+	if value == "" {
+		t.Fatalf("Variable %s not found in environment!", env_var)
+	}
+	for _, list := range lists {
+		for i := range *list {
+			str := (*list)[i]
+			(*list)[i] = strings.Replace(str, "$"+env_var, value, -1)
+		}
+	}
+}
+
+func ProcessCommand(t *testing.T, fullCommand []string, checkOutput bool) (string, string) {
+	var cmd *exec.Cmd
+	command := fullCommand[0]
+	flags := fullCommand[1:]
+	if len(flags) > 0 {
+		cmd = exec.Command(command, flags...)
+	} else {
+		cmd = exec.Command(command)
+	}
+
+	if checkOutput {
+		t.Logf("Executing: %s", cmd.Args)
+	} else {
+		t.Logf("Executing setup/teardown: %s", cmd.Args)
+	}
+
+	var outbuf, errbuf bytes.Buffer
+
+	cmd.Stdout = &outbuf
+	cmd.Stderr = &errbuf
+
+	if err := cmd.Run(); err != nil {
+		if checkOutput {
+			// The test might be designed to run a command that exits with an error.
+			t.Logf("Error running command: %s. Continuing.", err)
+		} else {
+			stdout := outbuf.String()
```

Comment:
Could you collapse this `else` block from here to 154 with the code at 156 to 163?